scraper/avecmondoc/avecmondoc.py

```diff
@@ -0,0 +1,392 @@
+        end_date = start_date + timedelta(days=AVECMONDOC_CONF.get("slot_limit", 50))
+        request.add_vaccine_type(get_vaccine_name(reason["reason"]))
+        availabilities = get_availabilities(reason["id"], reason["organizationId"],
+            start_date, end_date, client)
+        date, appointment_count = parse_availabilities(availabilities)
+        request.appointment_count += appointment_count
+        for appointment_schedule in appointment_schedules:
+            s_date = isoparse(appointment_schedule["from"])
+            n_date = isoparse(appointment_schedule["to"])
+            name = appointment_schedule["name"]
+            if name == "chronodose" and get_vaccine_name(reason["reason"]) not in CHRONODOSES["Vaccine"]:
+                continue
+            appointment_schedule["total"] += count_appointements(availabilities, s_date, n_date)
+        request.appointment_schedules = appointment_schedules
+        if date is None:
+            continue
+        if first_availability is None or first_availability > date:
+            first_availability = date
+    
+    return(first_availability)
+
+
+def center_to_centerdict(center: CenterInfo) -> dict:
+    center_dict = {}
+    center_dict["rdv_site_web"] = center.url
+    center_dict["nom"] = center.nom
+    center_dict["type"] = DRUG_STORE
+    center_dict["business_hours"] = center.metadata["business_hours"]
+    center_dict["phone_number"] = center.metadata["phone_number"]
+    center_dict["address"] = f'{center.metadata["address"]}, {center.location.cp} {center.location.city}' 
+    center_dict["long_coor1"] = center.location.longitude
+    center_dict["lat_coor1"] = center.location.latitude
+    center_dict["com_nom"] = center.location.city
+    center_dict["com_cp"] = center.location.cp
+    center_dict["com_insee"] = departementUtils.cp_to_insee(center.location.cp)
+    center_dict["gid"] = center.internal_id
+    return center_dict
+
+
+def center_iterator(client: httpx.Client = DEFAULT_CLIENT):
+    organization_slugs = []
+    search_result = search(client)
+    if search_result is None:
+        return []
+    if "data" not in search_result:
+        return []
+    for slug in search_result["data"]:
+        organizations = []
+        slug_type = slug["type"]
+        if slug_type == "doctor":
+            doctor_id = slug["id"]
+            for doctor_organization in get_by_doctor(doctor_id, client):
+                organizations.append(get_organization_slug(doctor_organization["slug"], client))
+        elif slug_type == "organization":
+            organizations = [get_organization_slug(slug["organizationSlug"], client)]
+        for organization in organizations:
+            if organization["zipCode"] is None or len(organization["zipCode"]) != 5:
+                continue
+            organization_slug = organization["slug"]
+            if organization_slug in organization_slugs:
+                continue
```

Comment:
@ValentinMouret j'aimerais bien ton avis sur l'utilisation des `continue`. On le fait beaucoup et je suis pas fan, mais j'ai peut être loupé la "meta" pour python

Reply:
J’essaye de limiter les `for` loops, parce que : 
* dans les 3/4 du temps une list comprehension est plus claire
* dans la moitié du quart qu’il reste, c’est pas une bonne idée de faire un for loop

Dans le huitième qui reste, le `continue` peut être utile, parce que justement c’est des for loops un peu plus compliquées.

Typiquement, je pense qu’ici c’est évitable.

`continue` en lui-même, je ne trouve pas ça fou puisque ça brise le flow, c’est plus de charge cognitive, donc j’essaye d’éviter.